jib-core/src/main/java/com/google/cloud/tools/jib/builder/AuthenticatePushStep.java

```diff
@@ -50,19 +49,21 @@ public Authorization call() throws ExecutionException, InterruptedException {
             String.format(DESCRIPTION, buildConfiguration.getTargetServerUrl()))) {
       RegistryCredentials registryCredentials = NonBlockingFutures.get(registryCredentialsFuture);
       String registry = buildConfiguration.getTargetServerUrl();
+
+      String credentialHelperSuffix = registryCredentials.getCredentialHelperUsed(registry);
+      Authorization authorization = registryCredentials.getAuthorization(registry);
+      if (credentialHelperSuffix == null || authorization == null) {
+        // If no credentials found, give a warning and return null.
+        buildConfiguration
+            .getBuildLogger()
+            .warn("No credentials could be retrieved for registry " + registry);
```

Comment:
I wonder if we might confuse users by saying this? Not something you need to fix in this PR, but could we say that describes that this will still work if the target is public?

Reply:
I think in no case would it be valid to not have credentials to push to a registry. Even if the image is public (on the pull side), it would still be private on the push side. Having an image that anyone can push to without credentials doesn't seem like a good idea.

Reply:
Haha, oh this is push .